w_common/lib/src/common/cache/cache.dart

```diff
@@ -353,7 +355,7 @@ class Cache<TIdentifier, TValue> extends Object with Disposable {
   ///
   /// If the [Cache] [isOrWillBeDisposed] then a [StateError] is thrown.
   Future<bool> applyToItem(
-      TIdentifier id, dynamic callback(Future<TValue> value)) {
+      TIdentifier id, dynamic callback(Future<TValue>? value)) {
     _log.finest('applyToItem id: $id');
     _throwWhenDisposed('applyToItem');
     if (_isReleased[id] != false) {
```

Comment:
I think it would be better not to allow this to call the callback with a `null` future.

```suggestion
      TIdentifier id, dynamic callback(Future<TValue> value)) {
    _log.finest('applyToItem id: $id');
    _throwWhenDisposed('applyToItem');
    if (_isReleased[id] != false || _cache[id] == null) {
```

Then change line 363 to 
```dart
final callBackResult = callback(_cache[id]!);
```